pkg/cluster/fixssh.go

```diff
@@ -33,9 +31,24 @@ func (m *manager) fixSSH(ctx context.Context) error {
 		return fmt.Errorf("unknown architecture version %d", m.doc.OpenShiftCluster.Properties.ArchitectureVersion)
 	}
 
-	lb, err := m.loadBalancers.Get(ctx, resourceGroup, lbName, "")
+	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	lb, err := m.checkAndUpdateLB(ctx, resourceGroup, lbName)
 	if err != nil {
-		return err
+		m.log.Warnf("Failed checking and Updating Load Balancer with err %s", err)
+	}
+
+	err = m.checkandUpdateNIC(ctx, resourceGroup, infraID, lb)
+	if err != nil {
+		m.log.Warnf("Failed checking and Updating Network Interface with err %s", err)
+	}
+	return nil
+}
+
+func (m *manager) checkAndUpdateLB(ctx context.Context, resourceGroup string, lbName string) (lb mgmtnetwork.LoadBalancer, err error) {
```

Comment:
now that we have some more methods defined can you please add tests around the specific functionality they provide? Currently the tests in `fixssh_test.go` do not cover any scenarios where `fixssh()` would return an error.

Reply:
Thanks Spencer for your feedback. Now I added couple of tests where the fixssh() would return an error.